### PR TITLE
Network receiver

### PR DESCRIPTION
**Pull request by tumbili**

### CODE REVIEW

**Comment by tumbili** [comment]
@mcharleb @LorenzMeier I implemented this in order to receive data over the socket from QGC.
Unfortunately, as soon as I try to connect with QGC I get the following message:
WARN  node_open as advertiser failed. (file /home/roman/Firmware/src/modules/uORB/uORBManager_posix.cpp line 98)

File: src/modules/mavlink/mavlink_receiver.cpp [hunk 1]
```diff
@@ -1624,34 +1624,49 @@ MavlinkReceiver::handle_message_hil_state_quaternion(mavlink_message_t *msg)
...
+	fds[0].events = POLLIN;
+#endif
 
 	ssize_t nread = 0;
 
 	while (!_mavlink->_task_should_exit) {
-		if (poll(fds, 1, timeout) > 0) {
-
+		if (poll(&fds[0], 1, timeout) > 0) {
+#ifndef __PX4_POSIX
```
Comment (LorenzMeier) [inline src/modules/mavlink/mavlink_receiver.cpp]: We still have serial on POSIX. This should not be an ifdef, but a check wether UDP or serial was selected

File: src/modules/mavlink/mavlink_receiver.cpp [hunk 2]
```diff
@@ -1624,34 +1624,49 @@ MavlinkReceiver::handle_message_hil_state_quaternion(mavlink_message_t *msg)
...
 			/* non-blocking read. read may return negative values */
 			if ((nread = ::read(uart_fd, buf, sizeof(buf))) < (ssize_t)sizeof(buf)) {
 				/* to avoid reading very small chunks wait for data before reading */
 				usleep(1000);
 			}
+#endif
 
+#ifdef __PX4_POSIX
```
Comment (LorenzMeier) [inline src/modules/mavlink/mavlink_receiver.cpp]: Same here, ifdefs are the wrong approach.

File: src/modules/mavlink/mavlink_receiver.cpp [hunk 3]
```diff
@@ -1663,11 +1678,12 @@ MavlinkReceiver::receive_thread(void *arg)
 				}
 			}
 
+#ifndef __PX4_POSIX
```
Comment (LorenzMeier) [inline src/modules/mavlink/mavlink_receiver.cpp]: Why are we not counting bytes on POSIX?

**Comment by tumbili** [comment]
Closing, changes are in PX4 master

**Pull request closed without merging**